docs/content/concepts/assets/software-defined-assets.mdx

```diff
@@ -0,0 +1,341 @@
+---
+title: Software-Defined Assets | Dagster
+description: A software-defined asset is a description of how to compute the contents of a particular data asset.
+---
+
+# Software-Defined Assets
+
+A software-defined asset is a description of how to compute the contents of a particular data asset.
+
+## Relevant APIs
+
+| Name                                  | Description                                                            |
+| ------------------------------------- | ---------------------------------------------------------------------- |
+| <PyObject object="asset" decorator /> | A decorator used to define assets.                                     |
+| <PyObject object="SourceAsset" />     | A class that describes an asset, but doesn't define how to compute it. |
+| <PyObject object="AssetGroup" />      | A group of software-defined assets.                                    |
+
+## Overview
+
+An "asset" is an object in persistent storage, e.g. a table, a file, or a persisted machine learning model. A software-defined asset is a Dagster object that couples an asset to the function and upstream assets that are used to producing its contents. Software-defined assets enable a declarative approach to data management, in which your code is the source of truth on what data assets should exist and how those assets are computed.
```

Comment:
some grammatical errors here, `couples an asset to a function, and a set of upstream assets that are used to produce that function's contents.`